httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CacheKeyGenerator.java

```diff
@@ -94,18 +96,67 @@ public String generateKey(final HttpHost host, final HttpRequest request) {
         }
     }
 
-    private String getFullHeaderValue(final Header[] headers) {
-        if (headers == null) {
-            return "";
+    /**
+     * Returns all variant names contained in {@literal VARY} headers of the given message.
+     *
+     * @since 5.3
+     */
+    public static List<String> variantNames(final MessageHeaders message) {
+        if (message == null) {
+            return null;
+        }
+        if (!message.containsHeader(HttpHeaders.VARY)) {
```

Comment:
Could we possibly omit the check `if (!message.containsHeader(HttpHeaders.VARY))`? 
If I understand correctly, if there are no `VARY` headers present, the loop `for (final Iterator<Header> it = message.headerIterator(HttpHeaders.VARY); it.hasNext(); )` wouldn't iterate, making the explicit check redundant. Am I missing something?"

Reply:
@arturobernalg It is a very valid point. It is a trade-off to save one intermediate object allocation if no `Vary` headers are present, but may be it is not worth it. I will get rid of it.